onnx/hub.py

```diff
@@ -272,3 +273,87 @@ def load(
             )
 
     return onnx.load(cast(IO[bytes], BytesIO(model_bytes)))
+
+
+def download_model_with_test_data(
+    model: str,
+    repo: str = "onnx/models:main",
+    opset: Optional[int] = None,
+    force_reload: bool = False,
+    silent: bool = False,
+) -> Optional[str]:
+    """
+    Downloads a model along with test data by name from the onnx model hub and returns the directory to which the files have been extracted.
+
+    :param model: The name of the onnx model in the manifest. This field is case-sensitive
+    :param repo: The location of the model repo in format "user/repo[:branch]".
+        If no branch is found will default to "main"
+    :param opset: The opset of the model to download. The default of `None` automatically chooses the largest opset
+    :param force_reload: Whether to force the model to re-download even if its already found in the cache
+    :param silent: Whether to suppress the warning message if the repo is not trusted.
+    :return: str or None
+    """
+    selected_model = get_model_info(model, repo, opset)
+
+    local_model_with_data_path_arr = selected_model.metadata[
+        "model_with_data_path"
+    ].split("/")
+    if selected_model.metadata["model_with_data_sha"] is not None:
+        local_model_with_data_path_arr[
+            -1
+        ] = f"{selected_model.metadata['model_with_data_sha']}_{local_model_with_data_path_arr[-1]}"
+    local_model_with_data_path = join(
+        _ONNX_HUB_DIR, os.sep.join(local_model_with_data_path_arr)
+    )
+
+    if force_reload or not os.path.exists(local_model_with_data_path):
+        if not _verify_repo_ref(repo) and not silent:
+            msg = (
+                'The model repo specification "{}" is not trusted and may'
+                " contain security vulnerabilities. Only continue if you trust this repo."
+            ).format(repo)
```

Comment:
Using [f-string](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals) to represent is preferred in ONNX. I know there are some existing .format in the same file. If you have time, could you please change .format into f-string (including other .format uses in the same file)? Thank you!